src/util/formDevices.tsx

```diff
@@ -232,3 +232,36 @@ export const deduplicateName = (
   }
   return candidate;
 };
+
+export const addNicDevice = ({
+  formik,
+  deviceName,
+  deviceNetworkName,
+}: {
+  formik: InstanceAndProfileFormikProps;
+  deviceName: string;
+  deviceNetworkName: string;
+}) => {
+  const copy = [...formik.values.devices];
+  copy.push({
+    type: "nic",
+    name: deviceName,
+    network: deviceNetworkName,
+  });
+  formik.setFieldValue("devices", copy);
+};
+
+export const detachOverriddenNetwork = ({
+  formik,
+  deviceName,
+}: {
+  formik: InstanceAndProfileFormikProps;
+  deviceName: string;
+}) => {
+  const copy = [...formik.values.devices].filter((t) => t.name !== deviceName);
+  copy.push({
+    type: "none",
+    name: deviceName,
+  });
+  formik.setFieldValue("devices", copy);
+};
```

Comment:
I think we could enhance the `addNoneDevice` function above with the filter logic and use it instead. Then we'd not need this new function.